massa-grpc/src/service.rs

```diff
@@ -0,0 +1,138 @@
+        let mut svc = GrpcServer::new(self)
+            .max_decoding_message_size(config.max_decoding_message_size)
+            .max_encoding_message_size(config.max_encoding_message_size);
+
+        if let Some(encoding) = &config.accept_compressed {
+            if encoding.eq_ignore_ascii_case("Gzip") {
+                svc = svc.accept_compressed(CompressionEncoding::Gzip);
+            };
+        }
+
+        if let Some(encoding) = &config.send_compressed {
+            if encoding.eq_ignore_ascii_case("Gzip") {
+                svc = svc.send_compressed(CompressionEncoding::Gzip);
+            };
+        }
+
+        let (shutdown_send, shutdown_recv) = oneshot::channel::<()>();
+
+        let mut server_builder = tonic::transport::Server::builder()
+            .concurrency_limit_per_connection(config.concurrency_limit_per_connection)
+            .timeout(config.timeout)
+            .initial_stream_window_size(config.initial_stream_window_size)
+            .initial_connection_window_size(config.initial_connection_window_size)
+            .max_concurrent_streams(config.max_concurrent_streams)
+            .tcp_keepalive(config.tcp_keepalive)
+            .tcp_nodelay(config.tcp_nodelay)
+            .http2_keepalive_interval(config.http2_keepalive_interval)
+            .http2_keepalive_timeout(config.http2_keepalive_timeout)
+            .http2_adaptive_window(config.http2_adaptive_window)
+            .max_frame_size(config.max_frame_size);
+
+        if config.accept_http1 {
+            let mut router_with_http1 = server_builder
+                .accept_http1(true)
+                .layer(GrpcWebLayer::new())
+                .add_service(svc);
+
+            if config.enable_reflection {
+                let reflection_service = tonic_reflection::server::Builder::configure()
+                    .register_encoded_file_descriptor_set(FILE_DESCRIPTOR_SET)
+                    .build()?;
+
+                router_with_http1 = router_with_http1.add_service(reflection_service);
+            }
+
+            //  // TODO get config runtime
+            //  match self.cfg.tokio_runtime.take() {
+            //      Some(rt) => rt.spawn(self.start_inner(methods, stop_handle)),
+            //      None => tokio::spawn(self.start_inner(methods, stop_handle)),
+            //  };
+
+            tokio::spawn(
+                router_with_http1.serve_with_shutdown(config.bind, shutdown_recv.map(drop)),
+            );
+        } else {
+            let mut router = server_builder.add_service(svc);
+
+            if config.enable_reflection {
+                let reflection_service = tonic_reflection::server::Builder::configure()
+                    .register_encoded_file_descriptor_set(FILE_DESCRIPTOR_SET)
+                    .build()?;
+
+                router = router.add_service(reflection_service);
+            }
+
+            //  // TODO get config runtime
+            //  match self.cfg.tokio_runtime.take() {
```

Comment:
What about this ?

Reply:
we should be able to pass a custom RT to Tonic but in next version ? cc @modship